### PR TITLE
Connect recovery & forgot password

### LINKED ISSUE #78: connect recovery password
Connect frontend and backend part of:

- Forgot password;
- Recovery password;

### PR DESCRIPTION
**Pull request by maryanchuk04**

### CODE REVIEW

File: StudCity.Frontend/src/pages/RecoveryPassword/index.jsx [hunk 1]
```diff
@@ -21,49 +26,62 @@ function RecoveryPassword() {
...
-						className="h-full w-full object-cover m-auto " 
-					/>
-			<div className="w-1/2 bg-white  h-[85%] flex flex-col m-auto shadow-form rounded-xl absolute  -translate-x-2/4 -translate-y-2/4 top-1/2 left-1/2">
+	const handleSubmit = async (event) => {
+		event.preventDefault();
+
+		const isOK = await service.recoveryPassword({ id: accountId, password: formState.password });
+
+		if (isOK) {
+			setTimeout(() => navigate("/authenticate"), 2000);
```
Comment (maryanchuk04) [inline StudCity.Frontend/src/pages/RecoveryPassword/index.jsx]: Its SetTimeout for beautiful effect)

File: StudCity.Frontend/src/services/authenticateService.js [hunk 2]
```diff
@@ -68,10 +68,36 @@ export class AuthenticateService {
...
 			
 			showAlert(err.response.data.error, "error");
 			return false;
 		}
 	}
+
+	async forgotPassword(email) {
+		try {
+			const response = await this.service.get(`${this.authUrl}/forgot-password/${email}`);
+			response.status === 200 && showAlert("An email with a link to reset your password has been sent to your email", "success");
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/services/authenticateService.js]: it's better to use if statement here

File: StudCity.Frontend/src/services/authenticateService.js [hunk 3]
```diff
@@ -68,10 +68,36 @@ export class AuthenticateService {
...
 			}
 			
 			showAlert(err.response.data.error, "error");
 			return false;
 		}
 	}
+
+	async forgotPassword(email) {
+		try {
+			const response = await this.service.get(`${this.authUrl}/forgot-password/${email}`);
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/services/authenticateService.js]: and if you only use status from this api call, then I would suggest write it like this:
const { status } = await ....

File: StudCity.Frontend/src/services/authenticateService.js [hunk 4]
```diff
@@ -68,10 +68,36 @@ export class AuthenticateService {
...
+			response.status === 200 && showAlert("An email with a link to reset your password has been sent to your email", "success");
+		}
+		catch(err) {
+			showAlert(err.response.data.error, "error");
+		}
+	}
+
+	async recoveryPassword({ id, password }) {
+		try {
+			const response = await this.service.get(`${this.authUrl}/recovery-password/${id}/${password}`);
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/services/authenticateService.js]: here as well better to use destructuring

File: StudCity.Frontend/src/pages/RecoveryPassword/index.jsx [hunk 5]
```diff
@@ -21,49 +26,62 @@ function RecoveryPassword() {
...
+		}
+		else setTimeout(() => navigate("/forgot-password"), 2000);
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/RecoveryPassword/index.jsx]: use else after closing curly bracket of if statement

File: StudCity.Frontend/src/pages/ForgotPassword/index.jsx [hunk 6]
```diff
@@ -1,48 +1,56 @@
...
 					<ValidateTextField
-							placeholder="Enter your email"
-							required={true}
-							value={email}
-							withErrorMessage={true}
-							validator={emailValidator}
-							onChange={handleChangeEmail}
-						/>
+						placeholder="Enter your email"
+						required={true}
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/ForgotPassword/index.jsx]: you can remove curly brackets here and other places as well as they are not required here

File: StudCity.Frontend/src/pages/RecoveryPassword/index.jsx [hunk 7]
```diff
@@ -21,49 +26,62 @@ function RecoveryPassword() {
...
+						placeholder="Repeat password"
+						className=""
+						setDisabled={setDisabled}
+						onChange={handleChangeConfirmPassword}
+						value={formState.confirmPassword}
+						confirmationValue={formState.password}
+						validate={passwordMatchValidation}
+					/>
+					<Button disabled={disabled
+						|| formState.password.length <= 8
```
Comment (Serhiy-dotcom) [inline StudCity.Frontend/src/pages/RecoveryPassword/index.jsx]: consider optimizing this part of logic

**Review by Serhiy-dotcom — CHANGES_REQUESTED** [review]

File: StudCity.Frontend/src/pages/RecoveryPassword/index.jsx (hunk 5 above)
Comment (maryanchuk04) [reply StudCity.Frontend/src/pages/RecoveryPassword/index.jsx]: Go to ass!

**Review by Serhiy-dotcom — APPROVED** [review]
Good work!

**Review by maria2908 — APPROVED** [review]

**Pull request merged**